File: src/console/screen/ScreenManager.php
<?php

namespace pocketcloud\cloud\console\screen;

use pocketcloud\cloud\console\Console;
use pocketcloud\cloud\console\screen\impl\DefaultScreen;
use pocketcloud\cloud\Server;
use pocketcloud\cloud\util\misc\Tickable;
use pocketcloud\cloud\util\trait\SingletonTrait;
use RuntimeException;

final class ScreenManager implements Tickable {
    use SingletonTrait;

    private ?Screen $currentScreen = null;

    public function __construct() {
        self::setInstance($this);
    }

    public function tick(int $currentTick): void {
        $this->currentScreen?->tick($currentTick);
    }

    public function setCurrentScreen(Screen $currentScreen): void {
        $this->currentScreen?->onRemove($Server::getInstance()->tick);
        $this->currentScreen = $currentScreen;
        $this->currentScreen->initialize(Console::getInstance());
    }

    public function resetScreen(): void {
        $this->setCurrentScreen(new DefaultScreen());
    }

    public function getCurrentScreen(): Screen {
        return $this->currentScreen ?? throw new RuntimeException("Current screen cannot be null");
    }
}

File: src/main/java/de/pocketcloud/cloud/console/screen/ScreenManager.java
package de.pocketcloud.cloud.console.screen;

import de.pocketcloud.cloud.PocketCloud;
import de.pocketcloud.cloud.console.screen.impl.DefaultScreen;
import de.pocketcloud.cloud.tick.Tickable;
import lombok.Getter;
import lombok.experimental.Accessors;

public final class ScreenManager implements Tickable {

    @Getter
    @Accessors(fluent = true)
    private static ScreenManager instance = null;

    private Screen currentScreen = null;

    public ScreenManager() {
        instance = this;
    }

    @Override
    public void tick(long currentTick) {
        if (currentScreen == null) return;
        this.currentScreen.tick(currentTick);
    }

    public void set(Screen screen) {
        if (currentScreen != null) currentScreen.onRemove(PocketCloud.instance().currentTick());
        this.currentScreen = screen;
        screen.initialize(PocketCloud.instance().console());
    }

    public void reset() {
        currentScreen = new DefaultScreen();
    }

    public Screen get() {
        return currentScreen;
    }
}